src/keyboard.rs

```diff
@@ -0,0 +1,1343 @@
+    pub fn super_key(&self) -> bool {
+        self.intersects(Self::SUPER)
+    }
+}
+
+bitflags! {
+    /// Represents the current state of the keyboard modifiers
+    ///
+    /// Each flag represents a modifier and is set if this modifier is active.
+    #[derive(Default)]
+    pub struct ModifiersState: u32 {
+        // left and right modifiers are currently commented out, but we should be able to support
+        // them in a future release
+        /// The "shift" key.
+        const SHIFT = 0b100 << 0;
+        // const LSHIFT = 0b010 << 0;
+        // const RSHIFT = 0b001 << 0;
+        /// The "control" key.
+        const CONTROL = 0b100 << 3;
+        // const LCTRL = 0b010 << 3;
+        // const RCTRL = 0b001 << 3;
+        /// The "alt" key.
+        const ALT = 0b100 << 6;
+        // const LALT = 0b010 << 6;
+        // const RALT = 0b001 << 6;
+        /// This is the "windows" key on PC and "command" key on Mac.
+        const SUPER = 0b100 << 9;
+        // const LLOGO = 0b010 << 9;
+        // const RLOGO = 0b001 << 9;
+    }
+}
+
+#[cfg(feature = "serde")]
+mod modifiers_serde {
+    use super::ModifiersState;
+    use serde::{Deserialize, Deserializer, Serialize, Serializer};
+
+    #[derive(Default, Serialize, Deserialize)]
+    #[serde(default)]
+    #[serde(rename = "ModifiersState")]
+    pub struct ModifiersStateSerialize {
+        pub shift_key: bool,
+        pub control_key: bool,
+        pub alt_key: bool,
+        pub super_key: bool,
+    }
+
+    impl Serialize for ModifiersState {
+        fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+        where
+            S: Serializer,
+        {
+            let s = ModifiersStateSerialize {
+                shift_key: self.shift_key(),
+                control_key: self.control_key(),
+                alt_key: self.alt_key(),
+                super_key: self.super_key(),
+            };
+            s.serialize(serializer)
+        }
+    }
+
+    impl<'de> Deserialize<'de> for ModifiersState {
+        fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+        where
+            D: Deserializer<'de>,
+        {
+            let ModifiersStateSerialize {
+                shift_key,
+                control_key,
+                alt_key,
+                super_key,
+            } = ModifiersStateSerialize::deserialize(deserializer)?;
+            let mut m = ModifiersState::empty();
+            m.set(ModifiersState::SHIFT, shift_key);
+            m.set(ModifiersState::CONTROL, control_key);
+            m.set(ModifiersState::ALT, alt_key);
+            m.set(ModifiersState::SUPER, super_key);
+            Ok(m)
+        }
+    }
+}
+
+/// Contains the platform-native physical key identifier (aka scancode)
+#[derive(Clone, Copy, PartialEq, Eq, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+pub enum NativeKeyCode {
+    Unidentified,
+    Windows(u16),
+    MacOS(u32),
+    XKB(u32),
+}
```

Comment:
It should be `Unidentified` on the web, yes. Do you mean this should be stated in the documentation?

Reply:
> Do you mean this should be stated in the documentation?

I'd prefer that, yes.

Reply:
Why do we store the platform source? Just for debug info? These values should never be serialised between platforms, right?

Reply:
> These values should never be serialised between platforms, right?

Right, they shouldn't. But it's relatively easy to write a `Key` to a config file, then copy same config file to a different platform. If this would just be a `u32`, then maybe you had "delete file" bound to "F35" but on this platform it's now bound to the "Open Media Player" key.

Reply:
Does including the platform name actually fix this though — are the identifiers reliable on each platform?

Reply:
It fixes part of the issue. Of course if you swap out your keyboard using the same OS, Unidentified keys might still be swithced around. But just because we can't entirely solve the probem doesn't mean that we shouldn't try making the situation better.